src/librustc_codegen_ssa/mir/analyze.rs

```diff
@@ -44,29 +45,27 @@ pub fn non_ssa_locals<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
         }
     }
 
-    let (mir, non_ssa_locals) = analyzer.finalize();
-    fx.mir = Some(mir);
-    non_ssa_locals
+    analyzer.non_ssa_locals
 }
 
-struct LocalAnalyzer<'mir, 'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> {
+struct LocalAnalyzer<'mir, 'a, 'b, 'tcx, Bx: BuilderMethods<'a, 'tcx>> {
     fx: &'mir FunctionCx<'a, 'tcx, Bx>,
-    mir: &'a mut BodyCache<&'a Body<'tcx>>,
+    mir: &'b Body<'tcx>,
```

Comment:
I had to use a new lifetime here because otherwise the reference on `&mut BodyCache<...>` kept getting extended longer than it should have.